src/tests/generatekey.cpp

```diff
@@ -149,7 +151,10 @@ TEST_F(rnp_tests, rnpkeys_generatekey_testSignature)
 
                 /* Sign the file */
                 assert_true(cli_rnp_protect_file(&rnp));
-                close(pipefd[0]);
+                if (pipefd[0]) {
+                    close(pipefd[0]);
+                    pipefd[0] = 0;
```

Comment:
0 is a valid file descriptor in POSIX so it is not correct to use 0 here. While it is possible that standard input file descriptor is never closed in this context and never returned again as a pipe descriptor, but anyway -1 should be used instead. And all code pieces like `if (pipefd[0])` should be changed to `if (pipefd[0] != -1)`

Reply:
Thanks for letting me know. I have fixed that.